src/rpm/package.rs

```diff
@@ -240,24 +216,19 @@ impl RPMPackage {
             .get_entry_data_as_string(IndexSignatureTag::RPMSIGTAG_SHA256);
 
         if let Ok(md5) = md5 {
-            if md5 != recreated_from_content.header_and_content_digest {
+            if md5 != pkg_actual_digests.header_and_content_digest {
                 return Err(RPMError::DigestMismatchError);
             }
         }
 
         if let Ok(sha1) = sha1 {
-            if sha1 != recreated_from_content.header_digest {
+            if sha1 != pkg_actual_digests.header_digest_sha1 {
                 return Err(RPMError::DigestMismatchError);
             }
         }
 
         if let Ok(sha256) = sha256 {
-            let sha256_calculated = {
-                let mut hasher = sha2::Sha256::default();
-                hasher.update(self.content.as_slice());
```

Comment:
I was verifying the wrong thing here.  Existing tests don't catch it because EL7 predates SHA256HEADER